packages/xdl/src/start/ExpoUpdatesManifestHandler.ts

```diff
@@ -0,0 +1,148 @@
+import { getRuntimeVersionForSDKVersion } from '@expo/sdk-runtime-versions';
+import express from 'express';
+import http from 'http';
+import { parse } from 'url';
+import uuid from 'uuid';
+
+import {
+  Analytics,
+  Config,
+  ProjectAssets,
+  ProjectUtils,
+  resolveEntryPoint,
+  UrlUtils,
+} from '../internal';
+import {
+  getBundleUrlAsync,
+  getExpoGoConfig,
+  getPackagerOptionsAsync,
+  stripPort,
+} from './ManifestHandler';
+
+function getPlatformFromRequest(req: express.Request | http.IncomingMessage): string {
+  const url = req.url ? parse(req.url, true) : null;
+  const platform = url?.query.platform || req.headers['expo-platform'];
+  if (!platform) {
+    throw new Error('Must specify expo-platform header or query parameter');
+  }
+  return String(platform);
+}
+
+export async function getManifestResponseAsync({
+  projectRoot,
+  platform,
+  host,
+}: {
+  projectRoot: string;
+  platform: string;
+  host?: string;
+}): Promise<{
+  body: ExpoUpdatesManifest;
+  headers: Map<string, number | string | readonly string[]>;
+}> {
+  const headers = new Map<string, any>();
+  // set required headers for expo-updates manifest specification
+  headers.set('expo-protocol-version', 0);
+  headers.set('expo-sfv-version', 0);
+  headers.set('cache-control', 'private, max-age=0');
+  headers.set('content-type', 'application/json');
+
+  const hostname = stripPort(host);
+  const [projectSettings, bundleUrlPackagerOpts] = await getPackagerOptionsAsync(projectRoot);
+  const projectConfig = getConfig(projectRoot);
+  const entryPoint = resolveEntryPoint(projectRoot, platform, projectConfig);
+  const mainModuleName = UrlUtils.stripJSExtension(entryPoint);
+  const expoConfig = projectConfig.exp;
+  const expoGoConfig = await getExpoGoConfig({
+    projectRoot,
+    projectSettings,
+    mainModuleName,
+    hostname,
+  });
+
+  const runtimeVersion =
+    expoConfig.runtimeVersion ??
+    (expoConfig.sdkVersion ? getRuntimeVersionForSDKVersion(expoConfig.sdkVersion) : null);
+  if (!runtimeVersion) {
+    throw new Error('Must specify runtimeVersion or sdkVersion in app.json');
+  }
+
+  const bundleUrl = await getBundleUrlAsync({
+    projectRoot,
+    platform,
+    projectSettings,
+    bundleUrlPackagerOpts,
+    mainModuleName,
+    hostname,
+  });
+
+  // resolve all assets and set them on the manifest as { rawUrl?: string, assetKey?: string } objects
+  const assets = await ProjectAssets.resolveAndCollectExpoUpdatesManifestAssets(
+    projectRoot,
+    expoConfig,
+    path => bundleUrl!.match(/^https?:\/\/.*?\//)![0] + 'assets/' + path
+  );
+
+  const expoUpdatesManifest = {
+    id: uuid.v4(),
+    createdAt: new Date().toISOString(),
+    runtimeVersion,
+    launchAsset: {
+      key: mainModuleName,
+      contentType: 'application/javascript',
+      url: bundleUrl,
+    },
+    assets,
+    metadata: {}, // required for the client to detect that this is an expo-updates manifest
```

Comment:
we should change how we detect this in expo-updates; afaik this field is no longer used for anything else

Reply:
https://linear.app/expo/issue/ENG-1486/change-how-we-detect-eas-manifests-in-client-code